tests/test_converter.py

```diff
@@ -1176,6 +1176,66 @@ def test_qasm_(self, recorder):
         assert recorder.queue[1].parameters == []
         assert recorder.queue[1].wires == Wires([2, 0])
 
+        for i in range(2, 6):
+            assert recorder.queue[i].name == "MidMeasureMP"
+            assert recorder.queue[i].wires == Wires([i - 2])
+
+    def test_qasm_measure(self):
+        """Tests that measurements specified as an argument are added to the converted circuit."""
+        qasm_string = 'include "qelib1.inc";' "qreg q[2];" "creg c[2];" "h q[0];" "cx q[0], q[1];"
+        dev = qml.device("default.qubit")
+        measurements = [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))]
+        loaded_circuit = load_qasm(qasm_string, measurements=measurements)
+
+        # loaded circuit with measurements
+        @qml.qnode(dev)
+        def quantum_circuit1():
+            return loaded_circuit()
+
+        # native pennylane measurements
+        @qml.qnode(dev)
+        def quantum_circuit2():
+            load_qasm(qasm_string)()
+            return [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliZ(1))]
+
+        assert quantum_circuit1() == quantum_circuit2()
+
+    def test_qasm_mid_circuit_measure(self):
+        """Tests that the QASM primitive measure is correctly converted."""
+        qasm_string = (
+            'include "qelib1.inc";'
+            "qreg q[2];"
+            "creg c[2];"
+            "h q[0];"
+            "measure q[0] -> c[0];"
+            "rz(0.24) q[0];"
+            "cx q[0], q[1];"
+            "measure q -> c;"
+        )
+        dev = qml.device("default.qubit")
+        loaded_circuit = load_qasm(qasm_string)
+
+        # loaded circuit
+        @qml.qnode(dev)
+        def quantum_circuit1():
+            mid_measure, m0, m1 = loaded_circuit()
```

Comment:
The formatting checker in CI doesn't like unused variables - this or a similar update would fix it:

```suggestion
            mid_measure, _, m1 = loaded_circuit()
```